src/Shared/Components/PrerenderComponentApplicationStore.cs

```diff
@@ -23,33 +25,105 @@ public PrerenderComponentApplicationStore(string existingState)
                 throw new ArgumentNullException(nameof(existingState));
             }
 
-            ExistingState = JsonSerializer.Deserialize<Dictionary<string, byte[]>>(Convert.FromBase64String(existingState)) ??
-                throw new ArgumentException(nameof(existingState));
+            DeserializeState(Convert.FromBase64String(existingState));
+        }
+
+        protected void DeserializeState(byte[] existingState)
+        {
+            var state = JsonSerializer.Deserialize<Dictionary<string, byte[]>>(existingState);
+            if (state == null)
+            {
+                throw new ArgumentException("Could not deserialize state correctly", nameof(existingState));
+            }
+
+            var stateDictionary = new Dictionary<string, ReadOnlySequence<byte>>();
+            foreach (var (key, value) in state)
+            {
+                stateDictionary.Add(key, new ReadOnlySequence<byte>(value));
+            }
+
+            ExistingState = stateDictionary;
```

Comment:
Had to do this because System.Text.Json doesn't support serializing/deserializing to `ReadOnlySequence<byte>`. An alternative here could be to deserialize the dictionary using the lower level APIs.